app/Services/Model/Imp/ScheduleService.php

```diff
@@ -120,6 +120,20 @@ public function publishProposedActivityToSource(
 
                 if (!$selection_plan->isAllowProposedSchedules())
                     throw new ValidationException("selection plan id {$selection_plan->getId()} does not allow proposed schedules");
+
+                if (!is_null($schedule)) {
+                    $prevProposedScheduledEvent = $schedule->getProposedPublishedEventBeforeThan($event->getStartDate(), $event->getLocation());
+
+                    if (($event->getStartDate()->getTimestamp() - $prevProposedScheduledEvent->getEndDate()->getTimestamp()) / 60 < $event->getTrackTransitionTime())
+                        throw new ValidationException(
+                            "There must be a transition time of at least {$event->getTrackTransitionTime()} minutes between the end of the previous event ({$prevProposedScheduledEvent->getSummitEventId()}) and the start of the current one ({$event->getId()})");
+
+                    $nextEvent = $schedule->getProposedPublishedEventAfterThan($event->getEndDate(), $event->getLocation());
+
+                    if (($nextEvent->getStartDate()->getTimestamp() - $event->getEndDate()->getTimestamp()) / 60 < $event->getTrackTransitionTime())
```

Comment:
u need to check here if  nextEvent is not null and $event->getTrackTransitionTime() is not null

Reply:
done